src/titiler/xarray/titiler/xarray/io.py

```diff
@@ -66,55 +52,8 @@ def xarray_open_dataset(  # noqa: C901
 
     else:
         assert zarr is not None, "'zarr' must be installed to read Zarr dataset"
-
         xr_engine = "zarr"
-
-    if protocol in ["", "file"]:
-        filesystem = fsspec.filesystem(protocol)  # type: ignore
-        file_handler = (
-            filesystem.open(src_path)
-            if xr_engine == "h5netcdf"
-            else filesystem.get_mapper(src_path)
-        )
-
-    elif protocol == "s3":
-        assert (
-            s3fs is not None
-        ), "'aiohttp' must be installed to read dataset stored online"
-
-        s3_filesystem = s3fs.S3FileSystem()
-        file_handler = (
-            s3_filesystem.open(src_path)
-            if xr_engine == "h5netcdf"
-            else s3fs.S3Map(root=src_path, s3=s3_filesystem)
-        )
-
-    elif protocol == "gs":
-        assert (
-            gcsfs is not None
-        ), "'gcsfs' must be installed to read dataset stored in Google Cloud Storage"
-
-        gcs_filesystem = gcsfs.GCSFileSystem()
-        file_handler = (
-            gcs_filesystem.open(src_path)
-            if xr_engine == "h5netcdf"
-            else gcs_filesystem.get_mapper(root=src_path)
-        )
-
-    elif protocol in ["http", "https"]:
-        assert (
-            aiohttp is not None
-        ), "'aiohttp' must be installed to read dataset stored online"
-
-        filesystem = fsspec.filesystem(protocol)  # type: ignore
-        file_handler = (
-            filesystem.open(src_path)
-            if xr_engine == "h5netcdf"
-            else filesystem.get_mapper(src_path)
-        )
-
-    else:
-        raise ValueError(f"Unsupported protocol: {protocol}, for {src_path}")
```

Comment:
It's a lot simpler to rely on fsspec's import error message and use fsspec.filesystem(protocol). This also provides the benefit of supporting a wider range of protocols. Still, let me know if you'd rather control the error message through the removed approach.

Reply:
I think @abarciauskas-bgse wrote most of this originally in titiler-xarray

Reply:
This approach seems better to me too @maxrjones  I didn't know about fsspec's import error message and `fsspec.filesystem(protocol)`